### ISSUE TITLE
Upload current bot source and server configuration

### ISSUE THREAD

**Issue by TriAce** [TriAce/Saber#2]
We have Saber running our Discord Directory and Community server's functionality. Currently, she has: 
*  a role system - where users type a role and press enter, without prefix and it assigns and removes roles in a specific channel.
*  a karaoke queue system - what we use for karaoke to print queue for large user-engaged musical events.

Both need improvements later, but we will stick to working on Discord Directory functionality for now.

For the role detection system:
* [ ]  Add the roles detection functionality to the channel: #discord-directory for users to add the "Discord Adventurers" role.
* [ ]  Add the roles detection functionality to the channel: #roles-guide for users to add the "Musical, Showtime, Engage, Exhibition, Welcome Party, Updates, Advertisements, Commissions, Giveaways, Special Discordians, Discord Adventurers" role. 
Add a way to add roles and remove these self-assigned roles from the system/print the list that's in there already. (Mod only)

For the Karaoke system: 
* [ ]  Tie the queue system to the #musical channel. Making queue commands available in there only. (Later we will build multi-channel functionality, as things grow and we need to balance out things)

[Discord-Directory-saber-70d8158251e8ec49bc46b1dee4e20d3c0aa754a4.zip](https://github.com/TriAce/Saber/files/1463803/Discord-Directory-saber-70d8158251e8ec49bc46b1dee4e20d3c0aa754a4.zip)

Think the attached upload is the most updated version, I have~ Enjoy! I'll update more issues/tasks shortly. :)

**Comment by TriAce** [comment]
**Here's the status of each issue/feature in the bot currently:**

Entry System version 1.0 - Lifetime on-entry server configuration
This is basically done. Just need to add a privacy note and bot OAuth2 for authorizing the user, as indicated in the "Hubspot CRM" issue, and we should be good to go.

Upload current bot source and server configuration 
Just need to upload the bot into Github properly since I pulled from the Gitlab repo. Forget about the rest tasks in that issue, for now.

Server submissions - Version 1.0
Basically, this is done, but we need to change the embeds to markdown listings and an approval mechanic, for staff.

User Authorization + Pushing data to Hubspot CRM from bot
This is not done yet, so let's get it done.

Rewards module - Version 1.0
I just need a partial development of this module. Just a referral rewards system.
